app/services/admin/users_import_service.rb

```diff
@@ -0,0 +1,100 @@
+module Admin
+  class UsersImportService
+    include ActiveModel::Model
+    attr_accessor :tempfile
+
+    def initialize(*)
+      super
+      self.batch_size = 100
+      initialize_role_cache
+    end
+
+    def call
+      if tempfile.nil?
+        errors.add(:tempfile, 'No file given')
+        return
+      end
+      import_csv
+    end
+
+    private
+
+    attr_accessor :role_cache, :batch_size
+
+    def initialize_role_cache
+      self.role_cache = Hash.new do |hash, key|
+        Admin::Role.find_by(name: key).tap do |role|
+          hash[key] = role
+          raise "Unknown role #{key}" unless role.present?
+        end
+      end
+    end
+
+    def import_csv
```

Comment:
Can you refactor this a little? You have ```if dups.present?``` twice there. which is a little confusing. maybe use a different method that will handle duplicates?